src/staticdata.c

```diff
@@ -1563,38 +1573,83 @@ static jl_value_t *strip_codeinfo(jl_method_t *m, jl_value_t *ci_, int isdef)
     return ret;
 }
 
+static void record_field_change(jl_value_t **addr, jl_value_t *newval)
+{
+    ptrhash_put(&field_replace, (void*)addr, newval);
+}
+
 static void strip_specializations_(jl_method_instance_t *mi)
 {
     assert(jl_is_method_instance(mi));
     jl_code_instance_t *codeinst = mi->cache;
     while (codeinst) {
         if (codeinst->inferred && codeinst->inferred != jl_nothing) {
-            codeinst->inferred = strip_codeinfo(mi->def.method, codeinst->inferred, 0);
-            jl_gc_wb(codeinst, codeinst->inferred);
+            if (jl_options.strip_ir) {
+                record_field_change(&codeinst->inferred, jl_nothing);
+            }
+            else if (jl_options.strip_metadata) {
+                codeinst->inferred = strip_codeinfo_meta(mi->def.method, codeinst->inferred, 0);
+                jl_gc_wb(codeinst, codeinst->inferred);
+            }
         }
         codeinst = jl_atomic_load_relaxed(&codeinst->next);
     }
+    if (jl_options.strip_ir) {
+        record_field_change(&mi->uninferred, NULL);
+        record_field_change((jl_value_t**)&mi->backedges, NULL);
+        record_field_change((jl_value_t**)&mi->callbacks, NULL);
+        record_field_change(&mi->specTypes, (jl_value_t*)jl_emptytuple_type);
```

Comment:
These don't seem to have anything to do with strip_ir, as they are stripping world-age and (one of) the identification keys by which we look up the code to execute.
```suggestion
```

Reply:
That's true; I guess my assumption is that if you are running with strip-ir you won't be changing any code and so don't need the backedges. I can certainly keep this info for now though.